packages/derived-wallet-ethereum/src/EIP1193DerivedPublicKey.ts

```diff
@@ -59,39 +59,43 @@ export class EIP1193DerivedPublicKey extends AccountPublicKey {
 
   verifySignature({ message, signature }: VerifySignatureArgs): boolean {
     const parsedSigningMessage = parseAptosSigningMessage(message);
-    if (
-      !parsedSigningMessage ||
-      !(signature instanceof EIP1193DerivedSignature)
-    ) {
+
+    if (!parsedSigningMessage) {
       return false;
     }
 
-    const { issuedAt, siweSignature } = signature;
-    const signingMessageDigest = hashValues([message]);
-
-    // Obtain SIWE envelope for the signing message
-    const envelopeInput = {
-      ethereumAddress: this.ethereumAddress,
-      signingMessageDigest,
-      issuedAt,
-    };
-
-    const siweMessage =
-      parsedSigningMessage.type === "structuredMessage"
-        ? createSiweEnvelopeForAptosStructuredMessage({
-            ...parsedSigningMessage,
-            ...envelopeInput,
-            chainId: 0, // TODO: use 0 does not really work, either way remove this once we rewrite the regular sign message to use the default wallet signMessage
-          })
-        : createSiweEnvelopeForAptosTransaction({
-            ...parsedSigningMessage,
-            ...envelopeInput,
-            chainId:
-              parsedSigningMessage.rawTransaction.rawTransaction.chain_id
-                .chainId,
-          });
-
-    const recoveredAddress = verifyEthereumMessage(siweMessage, siweSignature);
+    let messageBytes: Uint8Array | string;
+    // Handle structured message, i.e. a message signed withAptosSignMessageInput
+    if (parsedSigningMessage.type === "structuredMessage") {
+      if (!(signature instanceof EIP1193Signature)) return false;
```

Comment:
this was previously at the top.
any reason for it to be moved?

Reply:
Because it can be either an `EIP1193Signature` or `EIP1193DerivedSignature`

See https://github.com/aptos-labs/aptos-wallet-adapter/pull/572#discussion_r2082468779

Reply:
Oh I thought it was the same class (names are very similar)